Tech Debt (#33)
kachvame/wiki: vault/wiki.guidelines.tech-debt-and-run.md

---
id: tech-debt-and-run
title: Tech Debt and Run
desc: ''
updated: 1664965070011
created: 1664482252323
excerpt: 'Very advanced technique, to be used only in dire situations.'
---

> Tech Debt and Run is a very advanced technique which should only be attempted by
> Señor developers and only in dire situations. It involves suggesting and
> pitching absurd and overly complicated tools to your team with the hopes that
> they'll catch the bait and start using them, simply based off your previous
> experience as a Señor developer.

Reviewer: ```suggestion
Tech Debt and Run is a very advanced technique that should only be attempted by
Señor developers and only in dire situations. It involves suggesting and
pitching absurd and overly complicated tools to your team with the hopes that
they'll catch the bait and start using them, purely based on your previous
experience as a Señor developer.
```

> This will result in significant tech debt being accumulated over the next
> several months (or years), slowing the team down from developing new important
> features.

Reviewer: ```suggestion
The result is the accumulation of significant tech debt over the next several
months (or years), slowing the team down from developing new critical features
```

This is usually followed by a swift [[wiki.glossary.mvp]].

Some examples include:
- https://linkerd.io/
- https://tilt.dev/
- https://www.the-guild.dev/graphql/mesh

> Even if the team doesn't stick with these tools, it can still be considered a
> win, because of the time "invested" in researching, creating PoCs, and load
> testing them.

Reviewer: ```suggestion
Regardless of whether the team sticks with these tools, it can still be
considered a win considering the time "invested" in researching, creating PoCs
and load testing, etc.
```
